doc/sf/logaddexp.qbk

```diff
@@ -0,0 +1,139 @@
+logsumexp_performance<double>/256/real_time                3369 ns         3369 ns       206185
+logsumexp_performance<double>/512/real_time                6644 ns         6644 ns       104999
+logsumexp_performance<double>/1024/real_time              13171 ns        13171 ns        53049
+logsumexp_performance<double>/2048/real_time              26217 ns        26217 ns        26677
+logsumexp_performance<double>/4096/real_time              52289 ns        52289 ns        13293
+logsumexp_performance<double>/8192/real_time             104491 ns       104491 ns         6684
+logsumexp_performance<double>/16384/real_time            209023 ns       209022 ns         3344
+logsumexp_performance<double>/32768/real_time            427500 ns       426348 ns         1635
+logsumexp_performance<double>/65536/real_time            868480 ns       861568 ns          812
+logsumexp_performance<double>/131072/real_time          1720450 ns      1710749 ns          399
+logsumexp_performance<double>/262144/real_time          3413579 ns      3404083 ns          205
+logsumexp_performance<double>/524288/real_time          6847605 ns      6829615 ns          104
+logsumexp_performance<double>/1048576/real_time        13848953 ns     13777481 ns           52
+logsumexp_performance<double>/real_time_BigO              13.17 N         13.11 N
+logsumexp_performance<double>/real_time_RMS                   1 %             1 %
+logsumexp_performance<long double>/64/real_time             964 ns          962 ns       695192
+logsumexp_performance<long double>/128/real_time           1861 ns         1857 ns       387080
+logsumexp_performance<long double>/256/real_time           3531 ns         3530 ns       195978
+logsumexp_performance<long double>/512/real_time           7068 ns         7064 ns        99505
+logsumexp_performance<long double>/1024/real_time         14141 ns        14104 ns        49727
+logsumexp_performance<long double>/2048/real_time         27746 ns        27734 ns        24691
+logsumexp_performance<long double>/4096/real_time         56015 ns        55832 ns        12317
+logsumexp_performance<long double>/8192/real_time        112986 ns       112513 ns         6057
+logsumexp_performance<long double>/16384/real_time       224276 ns       223801 ns         3050
+logsumexp_performance<long double>/32768/real_time       446180 ns       446040 ns         1561
+logsumexp_performance<long double>/65536/real_time       910474 ns       908197 ns          788
+logsumexp_performance<long double>/131072/real_time     1815533 ns      1811141 ns          390
+logsumexp_performance<long double>/262144/real_time     3624313 ns      3613352 ns          193
+logsumexp_performance<long double>/524288/real_time     7205777 ns      7201394 ns           99
+logsumexp_performance<long double>/1048576/real_time   14511913 ns     14482980 ns           49
+logsumexp_performance<long double>/real_time_BigO         13.82 N         13.80 N
```

Comment:
Do you want to run the performance test and paste in your results? I presume I'm in the minority running that platform.

Reply:
I'm on the same platform.

Reply:
Just ran it: I get way better performance numbers even though I'm also on M1 pro:

```
~/math ❯❯❯ ./logsumexp                                                                                                                 logaddexp ✭ ✱ ◼
Unable to determine clock rate from sysctl: hw.cpufrequency: No such file or directory
2022-02-17T18:48:34-08:00
Running ./logsumexp
Run on (10 X 24.1259 MHz CPU s)
CPU Caches:
  L1 Data 64 KiB (x10)
  L1 Instruction 128 KiB (x10)
  L2 Unified 4096 KiB (x5)
Load Average: 2.11, 2.98, 2.97
-----------------------------------------------------------------------------------------------
Benchmark                                                     Time             CPU   Iterations
-----------------------------------------------------------------------------------------------
logsumexp_performance<float>/64/real_time                   381 ns          381 ns      1876902
logsumexp_performance<float>/128/real_time                  758 ns          758 ns       919176
logsumexp_performance<float>/256/real_time                 1512 ns         1512 ns       461658
logsumexp_performance<float>/512/real_time                 3108 ns         3107 ns       231660
logsumexp_performance<float>/1024/real_time                6039 ns         6039 ns       113364
logsumexp_performance<float>/2048/real_time               12068 ns        12068 ns        57951
logsumexp_performance<float>/4096/real_time               24141 ns        24141 ns        28999
logsumexp_performance<float>/8192/real_time               48300 ns        48296 ns        14502
logsumexp_performance<float>/16384/real_time              96602 ns        96597 ns         7124
logsumexp_performance<float>/32768/real_time             193835 ns       193716 ns         3624
logsumexp_performance<float>/65536/real_time             391079 ns       390490 ns         1809
logsumexp_performance<float>/131072/real_time            788897 ns       786846 ns          881
logsumexp_performance<float>/262144/real_time           1576679 ns      1572358 ns          439
logsumexp_performance<float>/524288/real_time           3096897 ns      3096596 ns          223
logsumexp_performance<float>/1048576/real_time          6230387 ns      6226829 ns          111
logsumexp_performance<float>/real_time_BigO                5.94 N          5.94 N
logsumexp_performance<float>/real_time_RMS                    1 %             1 %
logsumexp_performance<double>/64/real_time                  381 ns          381 ns      1860810
logsumexp_performance<double>/128/real_time                 766 ns          766 ns       912914
logsumexp_performance<double>/256/real_time                1528 ns         1527 ns       453501
logsumexp_performance<double>/512/real_time                3047 ns         3045 ns       228287
logsumexp_performance<double>/1024/real_time               6048 ns         6047 ns       113350
logsumexp_performance<double>/2048/real_time              12079 ns        12079 ns        57846
logsumexp_performance<double>/4096/real_time              24146 ns        24146 ns        28979
logsumexp_performance<double>/8192/real_time              48430 ns        48400 ns        14498
logsumexp_performance<double>/16384/real_time             97772 ns        97628 ns         7238
logsumexp_performance<double>/32768/real_time            194687 ns       194544 ns         3580
logsumexp_performance<double>/65536/real_time            389308 ns       389047 ns         1788
logsumexp_performance<double>/131072/real_time           775900 ns       775851 ns          894
logsumexp_performance<double>/262144/real_time          1550739 ns      1550722 ns          450
logsumexp_performance<double>/524288/real_time          3098859 ns      3098655 ns          226
logsumexp_performance<double>/1048576/real_time         6203986 ns      6203777 ns          112
logsumexp_performance<double>/real_time_BigO               5.92 N          5.92 N
logsumexp_performance<double>/real_time_RMS                   0 %             0 %
```

Did you compile at `-O3`?

Reply:
I also get just a *bit* of performance out of:

```diff
diff --git a/include/boost/math/special_functions/logsumexp.hpp b/include/boost/math/special_functions/logsumexp.hpp
index 1a9fe38bc..69a5d21e6 100644
--- a/include/boost/math/special_functions/logsumexp.hpp
+++ b/include/boost/math/special_functions/logsumexp.hpp
@@ -21,25 +21,15 @@ Real logsumexp(ForwardIterator first, ForwardIterator last)
     using std::exp;
     using std::log1p;

-    const Real max_val = *std::max_element(first, last);
+    auto elem = std::max_element(first, last);
+    const Real max_val = *elem;

     Real result = 0;
-    bool skip_k = true; // Flag in case there are multiple of the same max value
     while (first != last)
     {
-        if (*first != max_val)
-        {
-            result += exp(*first - max_val);
-        }
-        else if (!skip_k)
-        {
-            ++result;
-        }
-        else
-        {
-            skip_k = false;
-        }
-
+       if (first != elem) {
+           result += exp(*first - max_val);
+       }
         ++first;
     }
```